lib/cli_and_music_library_controller.rb

```diff
@@ -0,0 +1,129 @@
+require 'colorize'
+class MusicLibraryController
+  attr_reader(:path)
+
+  def initialize(path = './db/mp3s')
+    MusicImporter.new(path).import
+  end
+
+  def call
+    puts cli_start
+    loop do
+      prompt
+      user_input = gets.chomp
+      break if user_input == 'exit'
+      if check_input(user_input)
+        send(cli_options[user_input]) 
+      else
+        undefined
+      end
+    end
+  end
+
+  def cli_start
+
+    "
+    =*=*=*=*=*=*=*=*=*=*=*=*=
+          Music Library
+    =*=*=*=*=*=*=*=*=*=*=*=*=
+    ".cyan
+  end
+
+  def cli_commands
+    puts "Commands:
+    1. list songs\t #lists all songs
+    2. list genres\t #lists all genres
+    3. list genre\t #lists a particular genre
+    4. list artist\t #lists a particular artist
+    5. play song\t #play a particular song
+    6. home\t #go to the home screen
+
+    exit\t #exit Music Library
+    ".green
+  end
+
+  def prompt
+    print '<Groove>'
+  end
+
+  def cli_options
+    {
+      'list songs' => :list_songs,
+      'list artists' => :list_artists,
+      'list genres' => :list_genres,
+      'play song' => :play_song,
+      'list artist' => :list_artist,
+      'list genre' => :list_genre,
+      'home' => :cli_commands
+    }
+  end
+
+  def check_input(input)
+    cli_options.include? input
+  end
+
+  def list_songs
+    Song.all.each_with_index do |song, index|
+      puts "#{index + 1}. #{song.artist.name} - #{song.name} - #{song.genre.name}"
+    end
+  end
+
+  def list_artists
+    Artist.all.each {|artist| puts artist.name.to_s}
+  end
+
+  def list_genres
+    Genre.all.each {|genre| puts genre.name.to_s}
+  end
+
+  def play_song
+    song = check_for_integer
+    if song
+      puts "Playing #{song.artist.name} - #{song.name} - #{song.genre.name}"
+    else
+      puts "The song number could not be found"
+    end
+  end
+
+  def list_genre
+    print 'Enter a genre to list '
+    genre = gets.chomp
+    genres = Genre.find_by_name(genre)
+    if genres
+      genres.songs.each do |song|
+        puts "#{song.artist.name} - #{song.name} - #{song.genre.name}"
+      end
+    else
+      puts "#{genre} not found"
+    end
+  end
+
+  def list_artist
+    print 'Enter an artist to list '
+    artist = gets.chomp
+    artists = Artist.find_by_name(artist)
+    if artists
+      artists.songs.each{ |song| puts "#{song.artist.name} - #{song.name} - #{song.genre.name}" }
```

Comment:
Separate `each` block with a space.
```ruby
artist.songs.each {...}
```
Better yet use a `do end` if the line surpasses 80 columns.